src/NServiceBus.TransportTests/NServiceBusTransportTest.cs

```diff
@@ -102,6 +103,8 @@ public void TearDown()
             var userName = GetUserName();
             await queueCreator.CreateQueueIfNecessary(queueBindings, userName);
 
+            await TransportInfrastructure.Start();
```

Comment:
This will break MSMQ since it adds diagnostics here 

https://github.com/Particular/NServiceBus.Transport.Msmq/blob/release-1.0/src/NServiceBus.Transport.Msmq/MsmqTransportInfrastructure.cs#L133

so the transport tests needs to add the diagnostics collection just like the endpointconfig does

https://github.com/Particular/NServiceBus/blob/release-7.0/src/NServiceBus.Core/EndpointConfiguration.cs#L32

yes it sucks :(

Reply:
👍 will add that

Reply:
If we want to exactly mimic the startup sequence as defined in https://github.com/Particular/NServiceBus/blob/release-7.0/src/NServiceBus.AcceptanceTests/Core/TransportSeam/When_initializing_transport.cs, then `Start` needs to be called earlier.

Looking at that test, I think we might have a few other things to rearrange slightly.